word_guess_owl.rb

```diff
@@ -0,0 +1,153 @@
+# Class to print out owl ASCII art for the game.
+# Could potentially be used to change difficulty level of game by limiting initial number of guesses
+class Owl
+  def initialize(initial_zz)
+    @initial_zz = initial_zz
+  end
+
+  attr_reader :initial_zz
+
+  def sleepy_owl
+    puts " {~;~}"
+    puts " /)__)"
+    puts "/  ;;"
+  end
+
+  def zzzz(wrong_guesses)
+    string = "\n     "
+    (@initial_zz - wrong_guesses).times do
+      string += "Z "
+    end
+    puts string
+  end
+
+  def awake_owl
+    puts " {O;O}"
+    puts " /)__)"
+    puts "/  ;;"
+  end
+
+  def winning
+    puts "\n     HOOOOO-RAY!"
+    puts " {^;^}"
+    puts " /)__)"
+    puts "/  ;;"
+  end
+end
+# Allows user to guess letters, validates their input and prompts again if not valid,
+# stores letters already guessed, and increments the number of wrong guesses,
+# generates blank letter spaces, and fills in the blanks as game progresses.
+class Game
+  def initialize (word)
+    @finished_word = word
+    @letters = word.upcase.chars
+    @wrong_guesses = 0
+    @guessed_letters = []
+    @word_array =@letters.length.times.collect {|x| "_"}
+  end
+
+  attr_reader :letters, :finished_word, :guessed_letters, :wrong_guesses, :word_array
+
+  def guess_letter
+    print "Please guess a letter: "
+    guess = gets.chomp.upcase
+    while @guessed_letters.include?(guess)
+      print "You already guessed that letter. Guess again: "
+      guess = gets.chomp.upcase
+    end
+    # only lets the user guess A through Z
+    guess_req = /[A-Z]/
+    # this is making sure that the user guess does not meet our regex requirements or more then one was put in
+    while !guess_req.match(guess) || guess.length > 1
+      print "That is not a correct input. Please guess a letter: "
+      guess = gets.chomp.upcase
+    end
+
+    if @letters.include?(guess)
+      @guessed_letters << guess
+      fill_in(guess)
+      return "That is a correct guess."
+    else
+      @guessed_letters << guess
+      @wrong_guesses += 1
+      return "That is a wrong guess."
+    end
+  end
+
+  def fill_in(guess)
+    @letters.each_with_index do |letter, i|
+      if letter == guess
+        @word_array[i] = guess
+      end
+    end
+    return @word_array
+  end
+
+  def print_word_array
+    printing_word = ""
+    @word_array.each do |letter|
+      printing_word += "#{letter} "
+    end
+    return printing_word
+  end
+
+  def what_was_guessed
+    user_guesses = ""
+    @guessed_letters.each do |letter|
+      user_guesses += letter + " "
+    end
+    return user_guesses
+  end
+end
+
+# generates random word for user to guess
+def word_generator
+  word_options = ["BAZAAR", "BROOD", "EYRIE", "HOOTING", "LOOMING", "NEST", "STOOPING", "BROOD", "DISS", "PAIR", "PARLIAMENT", "SAGACIOUSNESS", "STARE", "WISDOM", "STABLE", "PROHIBITION", "VOLERY", "BLIZZARD"]
```

Comment:
This means the word options are hard-coded in.  It would be advantageous to have them passed into `initialize` so you could create different games with different lists of words... small note.